tests/Strictness.hs

```diff
@@ -80,6 +80,15 @@ pInsertWithValueStrict f k v m
     | HM.member k m = isBottom $ HM.insertWith (const2 bottom) k v m
     | otherwise     = isBottom $ HM.insertWith f k bottom m
 
+pInsertWithKeyKeyStrict :: (Key -> Int -> Int -> Int) -> Int -> HashMap Key Int -> Bool
+pInsertWithKeyKeyStrict f v m = isBottom $ HM.insertWithKey f bottom v m
```

Comment:
Use `Fun (Key, Int, Int) Int`.

Reply:
Done, although it complained about instances for `Key` so I used `Int` as the first param and used `unK` to pull out the int. I can change to `Key` and add necessary instances if you'd like.

Reply:
Oops, thought these comments were for HashMapProperties, changing here as well.